app/src/main/java/net/squanchy/contest/ContestService.java

```diff
@@ -0,0 +1,39 @@
+package net.squanchy.contest;
+
+import net.squanchy.remoteconfig.RemoteConfig;
+import net.squanchy.service.firebase.FirebaseAuthService;
+import net.squanchy.service.firebase.FirebaseDbService;
+import net.squanchy.service.firebase.model.FirebaseAchievements;
+
+import io.reactivex.Single;
+import io.reactivex.functions.BiFunction;
+import io.reactivex.schedulers.Schedulers;
+
+public class ContestService {
+
+    private final RemoteConfig remoteConfig;
+    private final FirebaseDbService dbService;
+    private final FirebaseAuthService authService;
+
+    public ContestService(RemoteConfig remoteConfig, FirebaseDbService dbService, FirebaseAuthService authService) {
+        this.remoteConfig = remoteConfig;
+        this.dbService = dbService;
+        this.authService = authService;
+    }
+
+    public Single<ContestStandings> standings() {
+        return authService.ifUserSignedInThenObservableFrom(dbService::achievements)
+                .first(FirebaseAchievements.empty())
+                .zipWith(remoteConfig.contestGoal(), buildStandings())
```

Comment:
You should probably use `combineLatest()` instead:

>The `CombineLatest` operator behaves in a similar way to `Zip`, but while `Zip` emits items only when each of the zipped source `Observable`s have emitted a previously unzipped item, `CombineLatest` emits an item whenever _any_ of the source `Observable`s emits an item (so long as each of the source `Observable`s has emitted at least one item)

[_source_](http://reactivex.io/documentation/operators/combinelatest.html)